zkevm-circuits/src/copy_circuit.rs

```diff
@@ -714,34 +701,34 @@ pub mod dev {
             mut layouter: impl Layouter<F>,
         ) -> Result<(), halo2_proofs::plonk::Error> {
             let block = self.block.as_ref().unwrap();
-
-            let challenges =
-                Challenges::mock(Value::known(self.randomness), Value::known(self.randomness));
+            let challenge_values = config.1.values(&mut layouter);
 
             config
+                .0
                 .tx_table
-                .load(&mut layouter, &block.txs, self.max_txs, self.randomness)?;
-            config.rw_table.load(
+                .load(&mut layouter, &block.txs, self.max_txs, &challenge_values)?;
+
+            // TODO: RW table can't be passed `Challenges<Value<F>>` until the EVM circuit
```

Comment:
Let me reference this issue https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/659 so that we don't forget about this.